invenio_drafts_resources/records/api.py

```diff
@@ -33,57 +55,66 @@ class ParentRecord(RecordBase):
 
 
 class Record(RecordBase):
-    """Record base API.
+    """Record API."""
 
-    Note: This class is meant to work along a with a draft class.
-    """
+    #
+    # Configuration to be set by a subclass
+    #
 
-    # Configuration
+    #: The record's SQLAlchemy model class. Must be set by the subclass.
     model_cls = None
-
+    #: The parent state's SQLAlchemy model class. Must be set by the subclass.
+    versions_model_cls = None
+    #: The parent record's API class. Must be set by the subclass.
+    parent_record_cls = None
+
+    #
+    # System fields
+    #
+    #: The internal persistent identifier. Records and drafts share UUID.
     pid = PIDField('id', provider=DraftRecordIdProviderV2, delete=True)
 
+    #: System field to check if a record has been published.
     is_published = PIDStatusCheckField(status=PIDStatus.REGISTERED)
 
-    #: List of field names (strings) to copy from the draft on create.
-    create_from_draft = []
+    #: The parent record - the draft is responsible for creating the parent.
+    parent = ParentField(
+        ParentRecord, create=False, soft_delete=False, hard_delete=False)
 
-    #: List of field names (strings) to copy from the draft on update.
-    update_from_draft = []
+    #: Version relationship
+    versions = VersionsField(create=True, set_latest=True)
 
-    # TODO: Below three methods create_from(), update_from() and register() has
-    # to be refactored. They are accessing "parent" but it may not be defined.
-    # Instead, this work should be delegated to the system fields, but it's not
-    # easy to add a new pre/post_create_from pre/post_register hook.
     @classmethod
-    def create_from(cls, draft):
-        """Create a new record from a draft."""
-        record = cls.create(
-            {},
-            id_=draft.id,
-            pid=draft.pid,
-            **{f: getattr(draft, f) for f in cls.create_from_draft}
-        )
-
-        # NOTE: Merge pid into the current db session if not already in the
-        # session.
-        cls.pid.session_merge(record)
-        record.parent.__class__.pid.session_merge(record.parent)
-
-        record.update_from(draft)
-
+    def publish(cls, draft):
```

Comment:
Comment: I've moved this logic from ``service.publish()`` into here to ensure that it's clearly defined in the API class how a draft is published. This is because if e.g. ``parent`` or ``versions`` is left-out of the ``create()`` the versioning get's messed up.